src/VisualStudio/Core/Def/Venus/ContainedDocument.cs

```diff
@@ -184,14 +185,33 @@ public DocumentId FindProjectDocumentIdWithItemId(uint itemidInsertionPoint)
                 }
             }
 
+            // The requested insertion itemid wasn't found. Check if the containing
+            //   document matches the requested itemid.
+            var containingDocumentItemid = GetContainingDocumentItemId(hierarchy);
+            if (containingDocumentItemid == itemidInsertionPoint)
+            {
+                var containingDocument = GetOpenTextContainer().CurrentText.GetOpenDocumentInCurrentContextWithChanges();
+
+                return containingDocument.Id;
```

Comment:
There's already an Id property on this type you can return.

Reply:
Totally missed it, nice!